Content.Server/Storage/EntitySystems/StorageSystem.Fill.cs

```diff
@@ -39,8 +50,42 @@ private void OnStorageFillMapInit(EntityUid uid, StorageFillComponent component,
                 Logger.ErrorS("storage", $"Tried to StorageFill {entry.PrototypeId} inside {uid} but can't.");
                 EntityManager.DeleteEntity(ent);
             }
+        }
+
+        // handle orgroup spawns
+        foreach (var group in orGroupedSpawns)
+        {
+            double diceRoll = _random.NextDouble() * group.Value.CumulativeProbability;
+            _random.Shuffle(group.Value.Entries);
+
+            double cumulative = 0.0;
+            for (int i = 0; i < group.Value.Entries.Count; i++)
+            {
+                cumulative += group.Value.Entries[i].SpawnProbability;
+                if (diceRoll <= cumulative)
+                {
+                    for (var index = 0; index < group.Value.Entries[i].Amount; index++)
+                    {
+                        var ent = EntityManager.SpawnEntity(group.Value.Entries[i].PrototypeId, coordinates);
+
+                        if (storage.Insert(ent)) continue;
 
-            if (!string.IsNullOrEmpty(entry.GroupId)) alreadySpawnedGroups.Add(entry.GroupId);
+                        Logger.ErrorS("storage", $"Tried to StorageFill {group.Value.Entries[i].PrototypeId} inside {uid} but can't.");
+                        EntityManager.DeleteEntity(ent);
+                    }
+                    break;
+                }
+            }
+        }
+    }
+    private sealed class OrGroup
+    {
+        public List<EntitySpawnEntry> Entries { get; set; } = new();
+        public float CumulativeProbability { get; set; } = 0f;
+        public OrGroup(List<EntitySpawnEntry> entries, float cumulativeProbability)
+        {
+            Entries = entries;
+            CumulativeProbability = cumulativeProbability;
         }
```

Comment:
There's no point to having default values if you force people to provide them in a constructor anyways. Just get rid of the constructor.